ShibbolethAuthPlugin.inc.php

```diff
@@ -230,7 +231,7 @@ function isShibbolethConfigured() {
 	function handleRequest($hookName, $params) {
 		$page = $params[0];
 		$op = $params[1];
-		if ($this->getEnabled()
+		if ($this->getEnabled() && !($this->_isShibbolethOptional())
```

Comment:
I think this would read cleaner to create a variable for possible login operations and search across that, rather than duplicating the conditional logic and handler in the `else if`.  E.g.:
```
	// modify user login
	$loginOps = ['index', 'signIn', 'signOut'];
	if (!$this->_isShibbolethOptional()) {
		// If Shibboleth is required, override password functionality
		$loginOps = array_merge($loginOps, ['changePassword', 'lostPassword', 'requestResetPassword', 'savePassword']);
	}
	if ($this->getEnabled()
			&& ($page == 'shibboleth'
				|| ($page == 'login'
					&& array_search(
						$op,
						$loginOps
					))
				|| ($page == 'user'
					&& array_search(
						$op,
						array(
							'activateUser',
							'register',
							'registerUser',
							'validate',
						)
					)
				)
			)
		) {
```